server/utils/forms/inputs/twelveHourBritishDateTimeInput.ts

```diff
@@ -57,6 +60,35 @@ export default class TwelveHourBritishDateTimeInput {
       }
     }
 
+    if (this.earliestAllowedTime !== null && result < this.earliestAllowedTime) {
+      const isSameDate =
+        this.earliestAllowedTime.getDate() === result.getDate() &&
+        this.earliestAllowedTime.getMonth() === result.getMonth() &&
+        this.earliestAllowedTime.getFullYear() === result.getFullYear()
```

Comment:
Based on we don't use any of the hour/minute/second fields on `earliestAllowedTime`, would it be more appropriate to call it `earliestAllowedDate`? (And maybe error if non-zero time portions are given?)

Reply:
you can pass hour minute second fields in `earliestAllowedTime`, this is just checking whether or not to show the error in the date field or the time field by checking if the `earliestAllowedTime` is on the same calendar day as the input. see the unit tests for exampes.

Reply:
Ah, I see. The `result < this.earliestAllowedTime` is time-aware. Got it, thanks. I was expecting to see explicit hour/minute checks for some reason.